File: app/src/main/java/com/jakubwyc/mysecretvault/mapper/UserMapper.java
package com.jakubwyc.mysecretvault.mapper;


import com.jakubwyc.mysecretvault.model.User;
import com.jakubwyc.mysecretvault.repository.RealmUser;

public class UserMapper implements Mapper<User, RealmUser> {


    @Override
    public RealmUser map(User data) {
        RealmUser realmUser = new RealmUser();
        realmUser.login = data.getLogin();
        realmUser.passwordHash = data.getPasswordHash();
        return realmUser;
    }

}

File: app/src/main/kotlin/com/jakubwyc/mysecretvault/model/mapper/UserMapper.kt
package com.jakubwyc.mysecretvault.model.mapper


import com.jakubwyc.mysecretvault.model.User
import com.jakubwyc.mysecretvault.repository.RealmUser

class UserMapper : Mapper<User, RealmUser> {


    override fun map(data: User): RealmUser {
        val realmUser = RealmUser()
        realmUser.login = data.login
        realmUser.passwordHash = data.passwordHash
        return realmUser
    }

}
